chain/substreams/src/trigger.rs

```diff
@@ -271,3 +237,208 @@ where
+                entity: "Block".to_string(),
+                expected_new_value: "123".to_string(),
+            },
+            Case {
+                field: Field {
+                    name: "setting big int transactions count of 302".to_string(),
+                    value_type: FieldType::Bigint as i32,
+                    new_value: hex::decode("012e").unwrap(),
+                    new_value_null: false,
+                    old_value: vec![],
+                    old_value_null: true,
+                },
+                entity: "Block".to_string(),
+                expected_new_value: "302".to_string(),
+            },
+            Case {
+                field: Field {
+                    name: "setting big int transactions count of 209".to_string(),
+                    value_type: FieldType::Bigint as i32,
+                    new_value: hex::decode("00d1").unwrap(),
+                    new_value_null: false,
+                    old_value: vec![],
+                    old_value_null: true,
+                },
+                entity: "Block".to_string(),
+                expected_new_value: "209".to_string(),
+            },
+            Case {
+                field: Field {
+                    name: "setting big decimal value".to_string(),
+                    value_type: FieldType::Bigdecimal as i32,
+                    new_value: hex::decode("3133363633312e35").unwrap(),
+                    new_value_null: false,
+                    old_value: vec![],
+                    old_value_null: true,
+                },
+                entity: "Block".to_string(),
+                expected_new_value: "136631.5".to_string(),
+            },
+            Case {
+                field: Field {
+                    name: "setting big decimal value 2".to_string(),
+                    value_type: FieldType::Bigdecimal as i32,
+                    new_value: hex::decode("3133303730392e30").unwrap(),
+                    new_value_null: false,
+                    old_value: vec![],
+                    old_value_null: true,
+                },
+                entity: "Block".to_string(),
+                expected_new_value: "130709.0".to_string(),
+            },
+            Case {
+                field: Field {
+                    name: "setting big decimal value 3".to_string(),
+                    value_type: FieldType::Bigdecimal as i32,
+                    new_value: hex::decode("39373839322e36").unwrap(),
+                    new_value_null: false,
+                    old_value: vec![],
+                    old_value_null: true,
+                },
+                entity: "Block".to_string(),
+                expected_new_value: "97892.6".to_string(),
+            },
+        ];
+
+        for case in cases.into_iter() {
+            let value: Value = decode_entity_change(&case.field, &case.entity).unwrap();
+            let mut actual_value: String = String::new();
+            match case.field.value_type {
```

Comment:
you can use let actual_value = match ... and this way you don't need to define it as mut etc